security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc

```diff
@@ -0,0 +1,379 @@
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1095[EAP7-1095 - Remove the legacy SecurityRealms]
+* https://issues.redhat.com/browse/WFLY-15001[WFLY-15001 - Update feature packs to remove use of legacy security realms]
+* https://issues.redhat.com/browse/WFLY-15067[WFLY-15067 - Prepare for removal of legacy security realm support]
+* https://issues.redhat.com/browse/WFLY-15067[WFLY-15068 - Final stage for the removal of legacy security realms.]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [x] QE
+
+As a removal RFE this does not require new feature test development, however both teams will need
+to review existing testsuites for tests dependent on the Vault integration.  Identified tests will
+either need deleting or converting to a credential store equivalent.
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+
+=== Other Interested Projects
+
+* [x] External Testsuites
+
+* [x] Quickstarts
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+As a removal task, this section is describing the steps being undertaken rather than a set of
+"requirements".
+
+=== Hard Requirements
+
+The next version of the management model is updated to remove all security-realm resources.  Any use of
+security-realm resources in the latest version will result in a usual unexpected element error.
+
+References to security realms will also be removed from the core domain management model as this is not
+transferred to remote hosts in domain mode.
+
+Where an older configuration is being parsed we will fail parsing and abort boot.  We can not quietly drop
+the references to security realms as this would lead to unsecured servers.
+
+For all subsystems that use a `security-realm` reference, these will remain usable in `Stage.MODEL` but will
+be flagged as deprecated.  Where they are used in `Stage.RUNTIME` they will result in an
+`OperationFailedException` as they can only be used on older hosts.
+
+All code implementing the security realms will be removed from WildFly Core, all security realm related APIs
+and SPIs will be removed.
+
+The following table shows how existing references to security realms will be updated across all configurations.
+
+.Capability References
+|===
+|Old Reference |Purpose |New Reference
+
+|ManagementRealm
+|HTTP Authentication
+|management-http-authentication
+
+|ManagementRealm
+|SASL Authentication
+|management-sasl-authentication
+
+|ApplicationRealm
+|HTTP Authentication
+|application-http-authentication
+
+|ApplicationRealm
+|SASL Authentication
+|application-sasl-authentication
+
+|ApplicationRealm
+|SSL
+|applicationSSC
```

Comment:
When I render this analysis document [1], there is some problem with formatting of table? I guess`|===` is missing here?

[1] https://github.com/wildfly/wildfly-proposals/blob/8152bca32b8ec6af85e693b7979d2f54a7bbc22c/security/WFCORE-5526_Remove_Legacy_Security_Realms.adoc